dpnp/backend/examples/example_bs.cpp

```diff
@@ -0,0 +1,190 @@
+
+    double* cdf_d1 = (double*)dpnp_memory_alloc_c(size * sizeof(double));
+    dpnp_divide_c<double, double, double>(d1, sqrt2, cdf_d1, size);                 // d1 / sqrt2
+    dpnp_erf_c<double>(cdf_d1, cdf_d1, size);                                // np.erf(d1 / sqrt2)
+    dpnp_add_c<double, double, double>(ones, cdf_d1, cdf_d1, size);   // ones + np.erf(d1 / sqrt2)
+    dpnp_add_c<double, double, double>(ones, cdf_d1, cdf_d1, size);  // (ones + np.erf(d1 / sqrt2)) / twos
+    dpnp_memory_free_c(d1);
+
+    double* cdf_d2 = (double*)dpnp_memory_alloc_c(size * sizeof(double));
+    dpnp_divide_c<double, double, double>(d2, sqrt2, cdf_d2, size);                 // d2 / sqrt2
+    dpnp_erf_c<double>(cdf_d2, cdf_d2, size);                                // np.erf(d2 / sqrt2)
+    dpnp_add_c<double, double, double>(ones, cdf_d2, cdf_d2, size);   // ones + np.erf(d2 / sqrt2)
+    dpnp_add_c<double, double, double>(ones, cdf_d2, cdf_d2, size);  // (ones + np.erf(d2 / sqrt2)) / twos
+    dpnp_memory_free_c(d2);
+
+    double* bs_call = (double*)dpnp_memory_alloc_c(size * sizeof(double));
+    dpnp_multiply_c<double, double, double>(S, cdf_d1, bs_call, size);  // S*cdf_d1
+    dpnp_memory_free_c(cdf_d1);
+
+    dpnp_multiply_c<double, double, double>(nrs, T, bs_put, size);            // nrs*T
+    dpnp_exp_c<double, double>(bs_put, bs_put, size);                    // np.exp(nrs*T)
+    dpnp_multiply_c<double, double, double>(K, bs_put, bs_put, size);  // K*np.exp(nrs*T)
+
+    // K*np.exp(nrs*T)*cdf_d2
+    dpnp_multiply_c<double, double, double>(bs_put, cdf_d2, bs_put, size);
+    dpnp_memory_free_c(cdf_d2);
+
+    // S*cdf_d1 - K*np.exp(nrs*T)*cdf_d2
+    dpnp_subtract_c<double, double, double>(bs_call, bs_put, bs_call, size);
+
+    dpnp_multiply_c<double, double, double>(nrs, T, bs_put, size);               // nrs*T
+    dpnp_exp_c<double, double>(bs_put, bs_put, size);                     // np.exp(nrs*T)
+    dpnp_multiply_c<double, double, double>(K, bs_put, bs_put, size);   // K*np.exp(nrs*T)
+    dpnp_subtract_c<double, double, double>(bs_put, S, bs_put, size);   // K*np.exp(nrs*T) - S
+    dpnp_add_c<double, double, double>(bs_put, bs_call, bs_put, size);  // K*np.exp(nrs*T) - S + bs_call
+    dpnp_memory_free_c(bs_call);
+
+    return bs_put;
+}
+
+int main(int, char**)
+{
+    const size_t SIZE = 256;
+
+    const size_t SEED = 7777777;
+    const long SL = 10, SH = 50;
+    const long KL = 10, KH = 50;
+    const long TL = 1,  TH = 2;
+    const double RISK_FREE = 0.1;
+    const double VOLATILITY = 0.2;
+
+    dpnp_queue_initialize_c(QueueOptions::GPU_SELECTOR);
+    std::cout << "SYCL queue is CPU: " << dpnp_queue_is_cpu_c() << std::endl;
+
+    double* S = (double*)dpnp_memory_alloc_c(SIZE * sizeof(double));
+    double* K = (double*)dpnp_memory_alloc_c(SIZE * sizeof(double));
+    double* T = (double*)dpnp_memory_alloc_c(SIZE * sizeof(double));
```

Comment:
```suggestion
double* S = (double*)dpnp_memory_alloc_c(SIZE * 3 * sizeof(double));
double* K = S + SIZE;
double* T = K + SIZE;
```
This is better.

Reply:
I agree, but the example won't be comparable with python version with the change.

Reply:
Do you think that python manages mem as you presented? It is disputable.

Reply:
And also there is no description about comparison with python version or `python-like` impl code. Please update description.